src/index.tsx

```diff
@@ -81,12 +81,26 @@ export const useIntersectionObserver = ({
   const handleIntersect = (entries: IntersectionObserverEntry[]) => {
     if (!intersectObs) return
 
-    const [entry] = entries
-
-    if (callback && entry.isIntersecting) callback(entry)
-    if (options.triggerOnce && entry.isIntersecting) intersectObs.disconnect()
-
-    setInView(entry.isIntersecting)
+    // In the case that we only want to trigger once, we can consider
+    // that if any entry has intersected, we've scrolled past the observed
+    // element.
+    if (options.triggerOnce) {
+      const hasIntersected = entries.some(e => e.isIntersecting)
+      if (hasIntersected) {
+        callback?.(entries)
+        intersectObs.disconnect()
+      }
+
+      setInView(hasIntersected)
+      return
+    }
+
+    // Otherwise, in situations where scrolling is **really** fast or the browser
+    // is busy, we can consider that the last entry is the most up-to-date.
+    const isIntersecting = entries[entries.length - 1].isIntersecting
+    if (isIntersecting) callback?.(entries)
```

Comment:
So the callback will only be executed when the last entry is intersecting. I think this is fine, at least i cannot think of use cases where this would be problematic.

Reply:
Yeah, I couldn't find any definitive resource about whether the last entry is in fact the "latest" information, but my limited testing seems to show that it is.